source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -430,6 +434,66 @@ void InstanceImpl::PendingRequest::onFailure() {
 void InstanceImpl::PendingRequest::onRedirection(Common::Redis::RespValuePtr&& value,
                                                  const std::string& host_address,
                                                  bool ask_redirection) {
+  if (parent_.dns_cache_) {
+    resp_value_ = std::move(value);
+    ask_redirection_ = ask_redirection;
+    auto result =
+        parent_.dns_cache_->loadDnsCacheEntry(host_address, parent_.default_port_, false, *this);
+
+    cache_load_handle_ = std::move(result.handle_);
+    switch (result.status_) {
+    case Extensions::Common::DynamicForwardProxy::DnsCache::LoadDnsCacheEntryStatus::InCache: {
+      ASSERT(cache_load_handle_ == nullptr);
+      if (!result.host_info_.has_value() || !result.host_info_.value()->address()) {
+        auto host = host_;
+        onResponse(std::move(resp_value_));
+        host->cluster().stats().upstream_internal_redirect_failed_total_.inc();
```

Comment:
nit: you might want to add some debug logs for these error cases here and below.